modules/collect/src/test/java/com/opengamma/strata/collect/GuavateTest.java

```diff
@@ -180,6 +181,26 @@ public void test_ensureOnlyOne() {
     assertThrowsIllegalArg(() -> Stream.of("a", "b").reduce(Guavate.ensureOnlyOne()));
   }
 
+  //-------------------------------------------------------------------------
+  public void test_casting() {
+    assertEquals(Stream.empty().map(Guavate.casting(Integer.class)).collect(toList()), ImmutableList.of());
+    List<Number> baseList = Arrays.asList(1, 2, 3);
+    List<Integer> castList = baseList.stream().map(Guavate.casting(Integer.class)).collect(toList());
+    assertEquals(castList, ImmutableList.of(1, 2, 3));
```

Comment:
I'd prefer to have the assertion be `assertEquals(castList, baseList)` rather than writing out the contents of `baseList`  but happy to leave it.